src/wallet/components/WalletProvider.tsx

```diff
@@ -15,11 +22,33 @@ export type WalletProviderReact = {
 export function WalletProvider({ children }: WalletProviderReact) {
   const { chain } = useOnchainKit();
   const [isOpen, setIsOpen] = useState(false);
+  const [isClosing, setIsClosing] = useState(false);
+  const { address } = useAccount();
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = useCallback(() => {
+    if (!isOpen) {
+      return;
+    }
+
+    setIsClosing(true);
+    setTimeout(() => {
+      setIsOpen(false);
+      setIsClosing(false);
+    }, 300);
```

Comment:
you could watch the transitionEnd event instead of having to keep this delay and the animation in sync

Reply:
i've refactored to remove the timeouts and use onAnimationEnd instead